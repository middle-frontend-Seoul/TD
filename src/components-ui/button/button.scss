.button {
  cursor: pointer;
  padding: 12px 16px;
  z-index: 0;
  font-style: normal;
  font-weight: 500;
  box-sizing: border-box;
  text-align: center;
  outline: none;
  border: 1px solid;
  border-radius: 4px;
  transition: border 0.3s;

  &_size {
    &-small {
      min-width: 180px;
      height: 40px;
      font-size: $font-size-md;
      line-height: 18px;
    }
    &-medium {
      min-width: 250px;
      height: 58px;
      font-size: $font-size-xlg;
      line-height: 23px;
    }
    &-large {
      min-width: 320px;
      height: 62px;
      font-size: 23px;
      line-height: 25px;
    }
  }

  &_use {
    &-default {
      border: 1px solid;
      color: white;
      border-color: rgba(0, 0, 0, 0.7);
      background: rgba(0, 0, 0, 0.7);
      &:hover {
        border-color: white;
      }
    }
    &-primary {
      color: #354216;
      border-color: rgba(0, 0, 0, 0.7);
      background-color: white;
      &:hover {
        border-color: #354216;
      }
    }
  }
  &_radius-medium {
    border-radius: 29px;
  }
}
