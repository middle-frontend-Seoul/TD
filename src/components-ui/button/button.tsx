import React, { FC, ReactNode, CSSProperties } from 'react';
import cn from 'classnames';

import './button.scss';

export type ButtonUse = 'default' | 'primary';

export type ButtonSize = 'xsmall' | 'small' | 'medium' | 'large';

export interface IButtonProps {
  type?: 'button' | 'submit' | 'reset';
  className?: string;
  radius?: boolean;
  use?: ButtonUse;
  size?: ButtonSize;
  icon?: ReactNode;
  style?: CSSProperties;
  disabled?: boolean;
  loading?: boolean;
  onClick?: () => void;
  children: ReactNode;
}

export const Button: FC<IButtonProps> = ({
  className,
  children,
  disabled,
  onClick,
  radius,
  style,
  size = 'medium',
  type = 'button',
  use = 'default',
}) => {
  // method
  // ---------------
  const handleOnClick = () => {
    if (disabled) return;
    onClick?.();
  };

  // render
  // ---------------
  const classes = cn(
    className,
    'button',
    `button_use-${use}`,
    `button_size-${size}`,
    {
      [`button_radius-${size}`]: !!radius,
    }
  );

  return (
    <button
      type={type}
      style={style}
      className={classes}
      onClick={handleOnClick}
    >
      {children}
    </button>
  );
};
