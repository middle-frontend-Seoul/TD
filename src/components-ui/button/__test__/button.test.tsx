import React from 'react';
import renderer from 'react-test-renderer';

import { Button } from '../button';

it('renders button base', () => {
  const tree = renderer.create(<Button>test button</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders button size', () => {
  const tree = renderer.create(<Button size="xsmall">size</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
