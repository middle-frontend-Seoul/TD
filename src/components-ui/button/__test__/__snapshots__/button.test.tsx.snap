// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button base 1`] = `
<button
  className="button button_use-default button_size-medium"
  onClick={[Function]}
  type="button"
>
  test button
</button>
`;

exports[`renders button size 1`] = `
<button
  className="button button_use-default button_size-xsmall"
  onClick={[Function]}
  type="button"
>
  size
</button>
`;
